compiler/src/dotty/tools/dotc/cc/CaptureOps.scala

```diff
@@ -273,6 +268,30 @@ extension (tp: Type)
     case _ =>
       tp
 
+  /** The first element of this path type */
+  final def pathRoot(using Context): Type = tp.dealiasKeepAnnots match
+    case tp1: NamedType if tp1.symbol.owner.isClass => tp1.prefix.pathRoot
+    case ReachCapability(tp1) => tp1.pathRoot
```

Comment:
OR just do dealias, which drops all annotations. I am not sure we need additional cases for intersections. It would be hard to do: which side of an intersection is the prefix?

Reply:
Yes, I agree, we should just do dealias. For an intersection, we can call `pathRoot` on each side and return the side whose result is a `NamedType`?

Reply:
... or a `ThisType`. Yes that could work. But let's get a test case first where it matters. Right now I don't see it coming up.